src/planner.rs

```diff
@@ -97,26 +108,206 @@ impl ConexPlanner {
             .collect::<Vec<(String, Vec<ConexFile>)>>();
 
         // Pass 2: Split and collapse layers so the size is about 512MB.
-        // let mut new_layer_to_files = Vec::new();
-        // let mut current_layer_size: usize = 0;
-        // let mut new_layer = Vec::new();
-        // for (layer, files) in self.layer_to_files.iter() {
-        //     for file in files.iter() {
-        //         if current_layer_size + file.size > 512 * 1024 * 1024 {
-        //             new_layer_to_files.push((layer.to_owned(), new_layer.clone()));
-        //             new_layer = Vec::new();
-        //             current_layer_size = 0;
-        //         }
-
-        //         new_layer.push(file.to_owned());
-        //         current_layer_size += file.size;
-        //     }
-
-        //     if !new_layer.is_empty() {
-        //         new_layer_to_files.push((layer.to_owned(), new_layer));
-        //     }
-        // }
-
-        self.layer_to_files.clone()
+        let mut new_layer_to_files = Vec::new();
+        let mut current_layer_size: usize = 0;
+        let mut new_layer = Vec::new();
+        let mut num_layer = 0;
+        let mut total_size = 0;
+        for (layer, files) in self.layer_to_files.iter() {
+            for file in files.iter() {
+                total_size += file.size;
+                let mut segment_idx = 0;
+                let mut remainder_size = file.size;
+                while remainder_size != 0 {
+                    //println!("remainder size {}",remainder_size.to_string());
+                    let mut frag = file.clone();
+                    if frag.hard_link_to.is_none() || remainder_size + current_layer_size < self.split_threshold {
+                        if remainder_size != file.size {
+                            //Case where remainder is a leftover fragment
+                            frag.chunk_size = Some(remainder_size);
+                            frag.start_offset = Some(file.size - remainder_size);
+                            frag.segment_idx = Some(segment_idx);
+                        }
+                        new_layer.push(frag.to_owned());
+                        if frag.hard_link_to.is_some() {
+                            //soft links don't count
+                            current_layer_size += remainder_size;
+                        }
```

Comment:
let's test this. i don't think this will do it because we want any file that's a link to occupy 0 bytes of space of content. because both hard link and softlinks are pointers.